src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/cif/file/internal/CifFileResultCreator.java

```diff
@@ -0,0 +1,215 @@
+package pl.edu.icm.pl.mxrdr.extension.importer.cif.file.internal;
+
+import edu.harvard.iq.dataverse.importer.metadata.ResultField;
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import org.apache.commons.lang3.StringUtils;
+import pl.edu.icm.pl.mxrdr.extension.importer.MxrdrMetadataField;
+import pl.edu.icm.pl.mxrdr.extension.importer.ProcessingSoftwareMapper;
+import pl.edu.icm.pl.mxrdr.extension.importer.SymmetryStructureMapper;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class CifFileResultCreator {
```

Comment:
Possibly try approach similar to PDB `DatasetParser` instead, with more declarative feel to it.